kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkConfigurationServiceImpl.java

```diff
@@ -341,12 +339,36 @@ public synchronized ComponentConfiguration getConfiguration() throws KuraExcepti
         // This method returns the network configuration properties without the current values.
         // i.e. the ip address that should be applied to the system, but not the actual one.
         logger.debug("getConfiguration()");
-        return new ComponentConfigurationImpl(PID, getDefinition(),
-                getNetworkConfiguration().getConfigurationProperties());
+        Optional<NetworkConfiguration> networkConfiguration = getNetworkConfiguration(false);
+        if (networkConfiguration.isPresent()) {
+            return new ComponentConfigurationImpl(PID, getDefinition(),
+                    networkConfiguration.get().getConfigurationProperties());
+        } else {
+            throw new KuraRuntimeException(KuraErrorCode.CONFIGURATION_ERROR,
+                    "The network component configuration cannot be retrieved");
+        }
     }
 
     @Override
     public synchronized NetworkConfiguration getNetworkConfiguration() throws KuraException {
+        if (this.currentNetworkConfiguration.isPresent()) {
+            return this.currentNetworkConfiguration.get();
+        } else {
```

Comment:
is the else required here?

Reply:
This is the "old" method, now deprecated. I added the else branch to maintain the old behavior.

Reply:
True, but since we have the return in the if, we don't need the else, right?